### PR TITLE
Advanced options deadline

### PR DESCRIPTION
**Pull request by TheGreatGildo**
resolved some of the issues david mentioned.
pr1

### CODE REVIEW

File: public/locales/en.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "connectWallet": "Connect Wallet",
   "currencyReserves": "Currency reserves",
   "dailyVolume": "Daily volume",
+  "deadline": "Transaction Deadline",
```
Comment (alphastorm) [inline public/locales/en.json:20]: add these keys with null values to `public/locales/zh-CN.json` so we can add translations later

File: src/components/DeadlineField.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          placeholder="10"
+          onClick={(): PayloadAction<Deadlines> =>
+            dispatch(updateTransactionDeadline(Deadlines.Custom))
+          }
+          onChange={(e): void => {
+            let deadlineNumber
+            e.target.value === "" || parseFloat(e.target.value) < 0
+              ? // if left blank, deadline defaults to 10
+                (deadlineNumber = Deadlines.Ten)
+              : (deadlineNumber = parseFloat(e.target.value))
```
Comment (alphastorm) [inline src/components/DeadlineField.tsx:58]: can use ternary assignment more concisely:
```ts
    let deadlineNumber =
      e.target.value === "" || parseFloat(e.target.value) < 0
        ? Deadlines.Ten // if left blank, deadline defaults to 10
        : parseFloat(e.target.value)
```

File: src/components/ReviewSwap.tsx [hunk 3]
```diff
@@ -116,6 +117,10 @@ function ReviewSwap({ onClose, onConfirm, data }: Props): ReactElement {
               {formatSlippageToString(slippageSelected, slippageCustom)}%
             </span>
           </div>
+          <div className="row">
+            <span className="title">{t("deadline")}</span>
+            <span className="value floatRight">{transactionDeadline} min</span>
```
Comment (alphastorm) [inline src/components/ReviewSwap.tsx:122]: should use the translation for `min` here

File: src/pages/SwapBTC.tsx [hunk 4]
```diff
@@ -51,6 +52,7 @@ function SwapBTC(): ReactElement {
     },
     priceImpact: BigNumber.from("0"),
   })
+  // console.debug("WUBADUBALUBLUB", transactionDeadline)
```
Comment (alphastorm) [inline src/pages/SwapBTC.tsx:55]: remove this if unused

**Review by alphastorm — CHANGES_REQUESTED** [review]

File: public/locales/en.json (hunk 1 above)
Comment (TheGreatGildo) [reply public/locales/en.json:20]: added deadline and minutes null values to Chinese localization

File: src/components/DeadlineField.tsx (hunk 2 above)
Comment (TheGreatGildo) [reply src/components/DeadlineField.tsx]: changed as per suggested

File: src/components/ReviewSwap.tsx (hunk 3 above)
Comment (TheGreatGildo) [reply src/components/ReviewSwap.tsx]: changed to translation lib syntax for minutes

File: src/pages/SwapBTC.tsx (hunk 4 above)
Comment (TheGreatGildo) [reply src/pages/SwapBTC.tsx]: removed

File: src/components/DeadlineField.tsx [hunk 5]
```diff
@@ -51,12 +51,10 @@ export default function DeadlineField(): ReactElement {
             dispatch(updateTransactionDeadline(Deadlines.Custom))
           }
           onChange={(e): void => {
-            let deadlineNumber
-            e.target.value === "" || parseFloat(e.target.value) < 0
-              ? // if left blank, deadline defaults to 10
-                (deadlineNumber = Deadlines.Ten)
-              : (deadlineNumber = parseFloat(e.target.value))
-            dispatch(updateTransactionDeadline(deadlineNumber))
```
Comment (alphastorm) [inline src/components/DeadlineField.tsx]: don't think this was meant to be removed

Comment (TheGreatGildo) [reply src/components/DeadlineField.tsx]: reinserted

**Review by alphastorm — APPROVED** [review]

**Pull request merged**